### PR TITLE
V4 doesn't exist

### PR DESCRIPTION
**Pull request by donaltuohy**
# Why
v4 doesn't exist. It was updated manually by accident it seems.

# How
Revert to version used previously.

### CODE REVIEW

File: .github/workflows/release-cli-version.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ jobs:
             ev-${{ env.LINUX_TARGET }}-${{
             needs.get-version.outputs.full_version }}.tar.gz
       - name: Upload MacOS Release
-        uses: actions/upload-release-asset@v4
```
Comment (ghost) [inline .github/workflows/release-cli-version.yml:82]: ~this action needs to match the download action~

Comment (ghost) [reply .github/workflows/release-cli-version.yml:82]: wrong action but this one appears to be archived should we drop it?

Comment (donaltuohy) [reply .github/workflows/release-cli-version.yml:82]: It's not `actions/upload-artifact` it's `upload-release-asset`. Seems that restriction is only on `upload-artifact`?

Comment (donaltuohy) [reply .github/workflows/release-cli-version.yml:82]: > If you are uploading with actions/upload-artifact@v4, you must download with actions/download-artifact@v4.

reading here: https://github.blog/news-insights/product-news/get-started-with-v4-of-github-actions-artifacts/

Comment (ghost) [reply .github/workflows/release-cli-version.yml:82]: Yeah I miss read the name but `actions/upload-release-asset` appears to be archived

Comment (ghost) [reply .github/workflows/release-cli-version.yml:82]: docs indicate we should switch to https://github.com/softprops/action-gh-release

Comment (donaltuohy) [reply .github/workflows/release-cli-version.yml:82]: Switched to https://github.com/shogo82148/actions-upload-release-asset

**Review by ghost — APPROVED** [review]

**Pull request merged**